### ISSUE TITLE
Plastic credits: Plastic Credit Registry

### ISSUE THREAD

**Issue by SammyEmpower** [EmpowerPlastic/empowerchain#1039]
**Business objective**
Increase plastic credit funding

**Description**

**Intended outcome:**
Get more than100k in monthly Plastic Credit sales on Marketplace
Have 90% of collection orgs active on platform every month

**Comment by SammyEmpower** [comment]
@ba1ciu Is there any tickets that can connect to this epic?

**Comment by SammyEmpower** [comment]
This card is only the interface!
The registry is available in the backend  but the interface is needed.

**Comment by ba1ciu** [comment]
Old designs are here on the right: https://www.figma.com/file/WPcpOXkn1m9KEZfL9ql53I/B.-Empower-Platform?type=design&node-id=19738%3A2&mode=design&t=ZSsmzYDYWHDBhHH7-1

**Comment by SammyEmpower** [comment]
Public good
Customers are the ones who have a standard for plastic credit. 
Aim is to increase credibility of plastic credit market. Goal is that it should be decentralized where a DOW is available..
OBP (ocean bound plastic) would be interested in this?

**Question:** 
Direct buyer interface - Does the reseller or other need specific things?
What is the MVP of this registry UI?

Is there critical things missing for whoever we focus on?
The order of things is the question here. 
To make the exploration the bet would be crucial...
We have an idea of what we want to do but in exploration we would need to decide who is the most important one to solve this for in order to get this bet going. 

We would need to define this closer and having the user need in place would be helpful

**Comment by ba1ciu** [comment]
PC Certificate has a QR which should redirect the user to the "digital" version of the plastic credit tied to the certificate. It's not redirecting anywhere just yet, but seems like it should to the plastic credit page on the credit registry. So feels like the MVP could be just a page which loads up plastic credit data based on the certificate ID?

**Comment by SammyEmpower** [comment]
There is a Figma here: 
https://www.notion.so/empowerplastic/Plastic-Credit-Registry-5203814b6f584122a3831b98b66ddf8d
Unsure if the design is still relevant? 

There is also a Notion page here: https://www.notion.so/empowerplastic/Plastic-Credit-Registry-5203814b6f584122a3831b98b66ddf8d 
With further details. 

The closest we seem to be in terms of defining target customers for the registry are certification organizations such as OBP

```
Zero Plastic Oceans developed the Ocean Bound Plastic (OBP) Certification Program in collaboration with the certification group Control Union. This certification program aims to attest that: OBP plastic waste is collected adequately and ethically. OBP plastic waste is well managed once collected.
```

**Comment by ba1ciu** [comment]
Thanks @SammyEmpower, great kickstart!

Considerations in Notion are seriously outdated. We have seriously rethink features like _Top retireres_. Doesn't sound like mvp feature.
Also, you pasted Notion link instead of Figma one :)

**Comment by KingWilhelmX** [comment]
Please not redo the things we did a year ago,  why we have the registry etc
thats not doable... its for commercial purpose and I have been talking with
everyone about this for a long time, it is not guesswork and nonsense...

fre. 2. feb. 2024, 19:49 skrev Jakub Kobeldys ***@***.***>:

**Comment by ba1ciu** [comment]
I didn't mean that we have to redo stuff, because I think all of these features we talked about make sense. I think we actually probably have most of the work done by now. What I meant is that we should iron out the scope for release and what should be added later on.

**Comment by KingWilhelmX** [comment]
100% - I misunderstood 🙏

**Comment by gjermundgaraba** [comment]
@KingWilhelmX I think the first question we need to answer in the marketplace / registry meeting today is who is the first target customer (to help narrow down MVP scope)

**Comment by SammyEmpower** [comment]
For the collection org  - To display where their plastic credits are retired. 
What is needed for the release? Get to the smallest version of that. 

So the MVP version could include:
- Login options 
- Search bar (for finding projects or credit collections?)
- Map showing retired ( credits => Broken down on collection or project?) 
- Table showing the different (collections or projects?) 

**Some initial questions:** 
- Do we want this to be an open URL or do we require login? 
- If we want both, what is the difference when I as a collection org am logged in? Do i see only my retired credits? 
- Should the map and registry be shareable with a predefined search query? Ie: PlasticCollector A wants to share who has retired their credits
- Do we link the registry to a specific project or collection when one clicks on a specific "row" in the registry? (see the image below)
- Maybe we do not need to care as much about the target users but more about the target behavours? 



![Image](https://github.com/EmpowerPlastic/empowerchain/assets/132075311/2c19ca45-e37b-4725-80a8-2b37841952f2)



Let´s sit down and write down our value props. 
**End users**: Enterprises donate to get material collected and tracked on platform and then they use the material in their products.  (The empower end users)

**Comment by KingWilhelmX** [comment]
No login to the registry needed - it is a public registry and that's the main feature - we can make other access versions later for different use cases. 

Search bars and filtering views, map and list, yes sir!

**Comment by ba1ciu** [comment]
Would it be possible to have it ready to dev my mid next week? @SammyEmpower

**Comment by SammyEmpower** [comment]
Notes from the ZPO recording: 
Vincent: 
There are different use cases:
- People who are curious on specific projects
     - They want to see  what is happening on a specific project, how many credits have been retired and who they were sold to.
     - So basically they expect to be able to go into a project and see the details.
 - Then there is the case where you want to verify the claims of a compay.
     -  So if Coca cola says that their plastic is neutrallized and we have a serial number for a specific plastic batch. We should be able to search for that serial number and verify that it has been verified in the name of coca cola. So that what they are claiming is true.
     - You can present it chronologiaclly like in the design with basic information or you can click on buyer, perhaps we can get access to 2 interfaces? One where we can see everything that a buyer has purchased and neutralized and another where we can see what a project is issuing and retiring. The rest would be a simple list where we can search by any field. 
     - Willhelm: For example, filter on collection organizations, filter on buyers. The current list is just a timeline showing what has been retired. 
     
Vincent: A second stage could be a client view where we can see a summary of everything a client has done, a summary of the project. In the beginning this list is fine but in the furture those two interfaces would be good. 
The objective of a registry is to have information transparent for all and we can comply with that for the listing but then it is nice to have the other interfaces. 
Especially, the client side, we want to verify that the claims are correct. It is more useful for the general public if they can verify if a company is complying with what they are saying. The project view is for curious people who want to understand what the competition is doing. So the client view is more important. 

Loup: 
Resellers, are they mentioned in your registy? Do you mention that a credit has been issued to a specific project and then sold by a reseller or trader? 
Vincent: We mention who has retired the credit. So basically `Credit` is retired by `Name` on behalf of `Name` and then additional information.
So for example: Climeco has retired x credits on behalf of Coca cola for the plastic neutral bottle <- Which will be a line in their registry.  

The why of the retirement is not compulsory. One can do that anonomously as well.

**Comment by SammyEmpower** [comment]
Based on the discussions and the requirements from ZPO as well as our perspective of what the collector's use case is (view and share their credits which have been retired) the first MVP version of the registry includes, assuming we are optimizing for speed of development): 

**MVP**:
Design will need to be adjusted to reflect the MVP version
- Filters (Project name ([Combo box](https://mui.com/material-ui/react-select/ ) ), Location, Retire date, Collection org. name  ([Combo box](https://mui.com/material-ui/react-select/ ), Material ([Combo box](https://mui.com/material-ui/react-select/ ) ) <- These values reflect what we have in PCM listing page now:   
![image](https://github.com/EmpowerPlastic/empowerchain/assets/132075311/bcf83840-1bf5-4799-9ac7-e68612b38e9b)

- Button which saves the filtered results as a URL which can be shared 
- Map showing location of projects with retired credits (Taking inspiration from Atomler here) 
![image](https://github.com/EmpowerPlastic/empowerchain/assets/132075311/5259e364-1c3a-4a67-9922-8083ba7d24c8)
Map is impacted by the filters and is interactive see https://market.atomler.com/find
Our map MVP would not have any on click actions.

- Table showing the different projects (Project name, Total # of credits in project, # credits retired to date, total volume being saved, Location, Material, Retire date, Collection org. name)
Note: When filtering on project level, there will be several instances (rows if you will) of the same project but with different retire dates and buyers (sponsors) due to the possiblity of credits in a specific project being retired on different dates by different sponsors.  <- Another option, if we want to keep the first registry page "clean" could be to show the project name and total # of credits, total number of retired credits, total number of "saved material" and ignore the sponsors and retire dates in the registry table but then allow a visitor to click on a project in the registry and, as ZPO proposed, send them to another page where we have a list with rows (more detailed registry if you will) showing: # retrired credts, retire date and sponsor. This would then be a visual timeline for the project basically. 
- Sorting on name and weight


Possible features for a **Second iteration**:
- Search bar (add capability to search for custom serial numbers)
- Add on click action to map (showing an entry to a specific project page for example)
- Interface to view / verify the claims of a compay "_where we can see a summary of everything a company has done_"

Possible features for a **third iteration**
- Add _Top retireres._ 
- More?

Shall we adjust the design to reflect and clarify the MVP? @gjermundgaraba @ba1ciu

**Comment by gjermundgaraba** [comment]
Looping in @Len-Empower here ☝️

**Comment by SammyEmpower** [comment]
Added MVP mocks based on the iteration suggestions above;
![image](https://github.com/EmpowerPlastic/empowerchain/assets/132075311/65b9aa42-178b-48fb-9a16-6cba38a75047)
![image](https://github.com/EmpowerPlastic/empowerchain/assets/132075311/1c0f3b24-361f-44be-8e86-6ea58ee51721)

Figma: 
https://www.figma.com/file/WPcpOXkn1m9KEZfL9ql53I/B.-Empower-Platform?type=design&node-id=16368-19176&mode=design&t=tX6075vTwuKmOsDL-0

**Comment by SammyEmpower** [comment]
It could also make sense to have a project page, as mentioned yesterday and that page could be linked from both the marketplace as well as the registry. @LoupSales @ba1ciu @Len-Empower

**Comment by SammyEmpower** [comment]
Comments from meeting 2024-02-14:
Attending: @LoupSales @ba1ciu @jschill @SammyEmpower @Len-Empower

**First version**
- Use the A version 
- If the project page is ready, then include the possibility to click on a registry entry to go to the project page else, do not. 
- Ad Standard somewhere
- Add filter on sponsors

**Second version**
- Merge project page and registry into one? <- Play with the idea

**Comment by SammyEmpower** [comment]
There are a bunch of suggestions available in Figma now for the MVP version and the "second" iteration: 
![image](https://github.com/EmpowerPlastic/empowerchain/assets/132075311/c2fdee11-855f-42af-b8e0-4f7e375852e6)

**Comment by SammyEmpower** [comment]
New design and stricktly limited (intentionally) clickable flow available in Figma: 
https://www.figma.com/proto/WPcpOXkn1m9KEZfL9ql53I/B.-Empower-Platform?type=design&node-id=22455-20026&t=BYv0PUJddNd9i06R-0&scaling=min-zoom&page-id=16368%3A19176&starting-point-node-id=22455%3A20026

![image](https://github.com/EmpowerPlastic/empowerchain/assets/132075311/71e8e23d-8177-4122-b427-c30aed9341fe)

If we aim to get to market with the registry UI as soon as possible i would slice the cake as following:
Release 1: Registry UI with table, search filters and the share button
Release 2: Add the data page so each row in the registry UI is clickable
Release 3: Add the Map 
Release 4: Add the search bar  for ID

In pararell with working on Release 1 (or basically tomorrow) I would send out a survey to our resellers, collection orgs and the ones we label as B2C here:  https://docs.google.com/spreadsheets/d/1_ov48T5DJLnXZgnbxcymsXYPLhV5tEE_BNQfYFS54FQ/edit#gid=551703452

The purpose of the survey being
1. Which group the Registry UI is most valuable for at this point in time (in order to guide initial communication)
2. Understand what fields are most relevant to filter on (in order to make propper product decisions)

As for any promises done to stakeholders; If the outcome of a customer focused and data informed product development process deviates very much from what has been promised. It should be simple and ok to just say something like: "_We had that intention but learned that the large market has these prefrences which is why we decided to do some slight changes_"

@ba1ciu @KingWilhelmX

**Comment by jschill** [comment]
Now i havent seen the designs before so bare with me. Three questions:
1. Wouldnt it be an idea to use grid view with cards here instead of "list view"? it feels like the items are mostly the background and a bunch of semi empty columns? If the purpose is inspiration that is...
2. Are we using any ui framework doing this design? It will greatly improve development speed. If not please have a look a daisy (https://daisyui.com/) which is the one we have on the marketplace and its just a thin css layer that we can add without too much fuzz.
3. Please start with small screens, it will help to visualize whats important and it's always extremely time consuming doing the other way around (big screen-first approach).


@SammyEmpower FYI its not possible to click the screenshots. I think you did the same thing as i did when i tried to add the screenshots in the design review document.

**Comment by SammyEmpower** [comment]
1. The refrence is the original design in the Figma done by someone as seen below as well as this:  https://www.obpcert.org/registry/ 
[image](https://github.com/EmpowerPlastic/empowerchain/assets/132075311/51a14257-c3f0-45ea-bf51-5621ed400fc2)
Purpose not being inspirational but more of a place to verify that a purchased plastic credit really has removed plastic somewhere.

2. I took what was available since the purpose initially was to do prototypes to be tested and not final dev. ready designs and that there seemed to be stong feelings about not doing too much redesigning. However for the comboboxes, material design elements have been used: https://m3.material.io/components/menus/guidelines since i did not find any Figma components for those in the daisyui https://www.figma.com/community/file/1252306778489709049

3. Not done any smallscreen designs since the main usecase for registry is assumed to mainly be laptop. But as soon as we decide which version to bring into production we can do mobile versions for sure =) 

Thanks. I just copy paste the image into Github...So there might be something there. Need to see if it is a common problem. =)

**Comment by SammyEmpower** [comment]
@jschill it seems to work if you go edit the text, remove the ! in front of [image] save and then edit the text again, add the ! infront of [image] again and boom. 

See recording:

https://github.com/EmpowerPlastic/empowerchain/assets/132075311/e2cb24fa-6e44-4598-b25c-8e8ff26f68af

**Comment by SammyEmpower** [comment]
Datapage first version ready: 
![image](https://github.com/EmpowerPlastic/empowerchain/assets/132075311/1edbc7d9-ff8c-46c0-a87f-a6650a03e478)

https://www.figma.com/file/WPcpOXkn1m9KEZfL9ql53I/B.-Empower-Platform?type=design&node-id=16368-19176&mode=design&t=BYv0PUJddNd9i06R-0

**Comment by SammyEmpower** [comment]
Here is the video

Uploading www_screencapture_com_2024-2-22_09_25.mp4…


Animations of the map can be more fancy etc. but this should be enough to get some feedback :smiley: from Climeco perhaps, ZPO and some collector?

**Comment by SammyEmpower** [comment]
So strange! Thanks for the heads up @jschill 

It is availble here:  https://empowercrew.slack.com/archives/C04DXHL0F7U/p1708590607854569?thread_ts=1708527104.405069&cid=C04DXHL0F7U

**Comment by SammyEmpower** [comment]
Registry UI modified with "Retired by":
There are thus 2 clickable flows. One with "Sponsors" and one with "Retired by"

![image](https://github.com/EmpowerPlastic/empowerchain/assets/132075311/288254af-4abf-4bc2-a0fb-5e9f57dcafa9)

**Comment by SammyEmpower** [comment]
Figma found here: 
https://www.figma.com/file/WPcpOXkn1m9KEZfL9ql53I/B.-Empower-Platform?type=design&node-id=16368-19176&mode=design&t=PpHHotaXjwMm2qPp-0


**First version of registry UI**: 
Is for the usecase where someone has bought credits and wants to verify and check the plastic credits that they have bought or resold. 
To see that the credits bought are on the blockchain and are representing real plastic removal. So in terms of target customers, that would include resellers, a company like Coca-cola or individual buyers for example.

The scenarios for the new plastic credit retirements which show up in the registry UI are:

1. Automatic retirement (through current build of PCM)
2. Collector retires manually (through platform)
3. Reseller retires the plastic credit (no current UI support, can be done with API or CLI)

So the data we would get in the "Retired by" column (rename of Sponsor) would be:

1. The name which the buyer specifies in the certificate
2. <Collector> on behalf of <buyer>  (where we get the buyer from the UI input on Platform)
3. <Reseller/trader> on behalf of <buyer>  (Not yet specified how to get this info in)

How do we envision doing scenarion 3? 
Part of second iteration where target is more geard towards resellers? 

As for the old plastic credits: 
Sammy: Would we want to add the old plastic credit retirements to the Registry UI as well? Or are they stuck in another dimension?
Jakub: It is a bit of a different dimension. Probably doable, but would be a lot of prep work, like scripts, etc.

basically a need for decision on the old plastic credits is needed @gjermundgaraba
